test/Microsoft.IdentityModel.Tokens.Tests/CryptoProviderFactoryTests.cs

```diff
@@ -1118,19 +1118,96 @@ public void ProviderCacheTest_EnsureNoException_MultipleThreads()
+            var cache = CryptoProviderFactory.Default.CryptoProviderCache as InMemoryCryptoProviderCache;
+            Assert.True(cache.TaskCount == 0, $"ProviderCache_EnsureNoLeakingTasks_SecurityTokenHandler_SymmetricEncryptingCredentials: unexpected task count: {cache.TaskCount}, expected: 0");
+        }
+
+        /// <summary>
+        /// Test to ensure no hanging task at the end when calling the JwtSecurityTokenHandler.WriteToken() method.
+        /// The JwtHeder is created with SigningCredentials.
+        /// </summary>
+        [Fact]
+        public void ProviderCache_EnsureNoLeakingTasks_SecurityTokenHandler_SigningCredentials()
+        {
+            var testClaims = new List<Claim>
+            {
+                new Claim(ClaimTypes.AuthenticationMethod, Default.AuthenticationMethod, ClaimValueTypes.String, Default.Issuer),
+                new Claim(ClaimTypes.AuthenticationInstant, Default.AuthenticationInstant, ClaimValueTypes.DateTime, Default.Issuer)
+            };
+
+            var signingCredentials = new SigningCredentials(new X509SecurityKey(KeyingMaterial.DefaultCert_2048), SecurityAlgorithms.RsaSha256);
+
+            var token = new JwtSecurityToken(
+                issuer: "IssuerName",
+                audience: "Audience",
+                claims: testClaims,
+                notBefore: DateTime.Now.AddHours(-1),
+                expires: DateTime.Now.AddHours(1),
+                signingCredentials: signingCredentials);
+
+            string certHash = "Test Cert Hash";
+
+            token.Header[JwtHeaderParameterNames.X5t] = certHash;
+            token.Header[JwtHeaderParameterNames.Kid] = certHash;
+
+            var handler = new JwtSecurityTokenHandler();
+            _ = handler.WriteToken(token);
+
+            var cache = CryptoProviderFactory.Default.CryptoProviderCache as InMemoryCryptoProviderCache;
+            var waitTimeInSeconds = TaskStopWaitTimeInSeconds(cache.TaskExecutionTimeInSeconds);
+
+            WaitTillTasksStarted(cache, waitTimeInSeconds); // wait for the event queue task to start
+            Assert.True(cache.TaskCount == 1, $"ProviderCache_EnsureNoLeakingTasks_SecurityTokenHandler_SymmetricEncryptingCredentials: unexpected task count: {cache.TaskCount}, expected: 1");
+
+            WaitTillTasksComplete(cache, waitTimeInSeconds); // wait for the event queue task to complete
+            Assert.True(cache.TaskCount == 0, $"ProviderCache_EnsureNoLeakingTasks_SecurityTokenHandler_SigningCredentials: unexpected task count: {cache.TaskCount}, expected: 0");
+        }
+
+        /// <summary>
+        /// Calculate the wait time for a task to stop.
+        /// This is for adding more time allowing the task to exit properly.
+        /// </summary>
+        /// <param name="taskExecutionTimeInSeconds">the time the event queue task runs</param>
+        /// <returns>1.2 times of the taskExecutionTimeInSeconds</returns>
```

Comment:
Is there a reason for 1.2?

Reply:
No particular reason about the 'magic' number, but 1.2 as a factor seems reasonable. I added more comments to explain it.